Python/optimizer.c

```diff
@@ -1075,15 +1112,19 @@ prepare_for_execution(_PyUOpInstruction *buffer, int length)
                 exit_op = _HANDLE_PENDING_AND_DEOPT;
             }
             int32_t jump_target = target;
-            if (is_for_iter_test[opcode]) {
-                /* Target the POP_TOP immediately after the END_FOR,
-                 * leaving only the iterator on the stack. */
-                int extended_arg = inst->oparg > 255;
-                int32_t next_inst = target + 1 + INLINE_CACHE_ENTRIES_FOR_ITER + extended_arg;
-                jump_target = next_inst + inst->oparg + 1;
+            bool unique_target = false;
+            if (
+                opcode == _GUARD_IP__PUSH_FRAME ||
+                opcode == _GUARD_IP_RETURN_VALUE ||
+                opcode == _GUARD_IP_YIELD_VALUE ||
+                opcode == _GUARD_IP_RETURN_GENERATOR
+            ) {
+                exit_op = _DYNAMIC_EXIT;
+                unique_target = true;
             }
-            if (jump_target != current_jump_target || current_exit_op != exit_op) {
-                make_exit(&buffer[next_spare], exit_op, jump_target);
+            bool is_control_flow = (opcode == _GUARD_IS_FALSE_POP || opcode == _GUARD_IS_TRUE_POP || is_for_iter_test[opcode]);
```

Comment:
I think this is the only use of `is_for_iter_test`, in which case replace it with `is_control_flow` and simplify this test.

Reply:
No it's not. We also use it to decide to offset the target to pointer after the END_FOR too.